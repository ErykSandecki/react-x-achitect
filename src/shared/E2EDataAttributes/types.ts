// types
import { E2EAttribute } from 'types/e2e';

export type E2EType = E2EAttribute | string;
export type E2EValue = number | string;
