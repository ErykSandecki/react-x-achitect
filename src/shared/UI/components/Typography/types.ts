// types
import { TTypographyProps } from './Typography';

export type TOmittedTypographyProps = Omit<TTypographyProps, 'fontType'>;
