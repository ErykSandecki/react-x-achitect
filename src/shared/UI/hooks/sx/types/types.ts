// types
import { TSXSpacings } from './spacings';

export type TSX = {
  spacings?: TSXSpacings;
};
