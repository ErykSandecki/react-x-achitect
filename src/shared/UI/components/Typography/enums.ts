export enum TypographyColor {
  blue1 = 'blue1',
  green1 = '28a745',
  green2 = 'a7b45d',
  neutral1 = 'neutral1',
  neutral2 = 'neutral2',
  neutral3 = 'neutral3',
  neutral4 = 'neutral4',
  neutral5 = 'neutral5',
  orange1 = 'orange1',
  orange2 = 'orange2',
  pink1 = 'pink1',
  red1 = 'red1',
  skyBlue1 = 'skyBlue1',
  violet1 = 'violet1',
}

export enum TypographyFontStyle {
  italic = 'italic',
  normal = 'normal',
}

export enum TypographyFontType {
  h1 = 'h1',
  h2 = 'h2',
  h3 = 'h3',
  h4 = 'h4',
  h5 = 'h5',
  h6 = 'h6',
  p = 'p',
  small = 'small',
}

export enum TypographyFontWeight {
  blacks = 'blacks',
  bold = 'bold',
  extraBold = 'extra-bold',
  light = 'light',
  medium = 'medium',
  regular = 'regular',
  semiBold = 'semi-bold',
}
