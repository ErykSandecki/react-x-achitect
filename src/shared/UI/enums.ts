export enum InputSize {
  small = 'small',
  medium = 'medium',
  large = 'large',
}
