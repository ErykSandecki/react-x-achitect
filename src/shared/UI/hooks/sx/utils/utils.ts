import { kebabCase } from 'lodash';

// types
import { TSX } from '../types/types';

export const seperateCssProperties = (value: string): Array<string> =>
  value.split(';');

export const getCssStyles = (sx: TSX, keys: Array<string>): string =>
  keys
    .map((key) => {
      const value = sx[key];
      return value !== undefined ? `${kebabCase(key)}: ${value};` : '';
    })
    .filter(Boolean)
    .join('\n');
