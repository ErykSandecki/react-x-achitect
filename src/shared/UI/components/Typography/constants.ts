// others
import { COLORS } from 'constant/scss/variables/colors';

// types
import { Theme } from 'types/enums/theme';
import { TTypographyColorsMode } from './types';

export const TYPOGRAPHY_COLORS_MODE: TTypographyColorsMode = {
  link: {
    [Theme.dark]: COLORS.blue1Dark,
    [Theme.light]: COLORS.blue1Light,
  },
  neutral1: {
    [Theme.dark]: COLORS.neutral1Dark,
    [Theme.light]: COLORS.neutral1Light,
  },
  neutral2: {
    [Theme.dark]: COLORS.neutral2Dark,
    [Theme.light]: COLORS.neutral2Light,
  },
  neutral3: {
    [Theme.dark]: COLORS.neutral3Dark,
    [Theme.light]: COLORS.neutral3Light,
  },
  neutral4: {
    [Theme.dark]: COLORS.neutral4Dark,
    [Theme.light]: COLORS.neutral4Light,
  },
  neutral5: {
    [Theme.dark]: COLORS.neutral5Dark,
    [Theme.light]: COLORS.neutral5Light,
  },
} as const;
