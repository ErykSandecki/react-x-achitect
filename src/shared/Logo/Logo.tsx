import { FC, useContext } from 'react';
import { useHistory } from 'react-router';

// assets
import { ReactComponent as Light } from 'assets/images/logo--light.svg';
import { ReactComponent as Dark } from 'assets/images/logo-dark.svg';

// components
import E2EDataAttribute from 'shared/E2EDataAttributes/E2EDataAttribute';

// core
import { Context } from 'core/ContextProvider/ContextProvider';

// types
import { E2EAttribute } from '../../types/e2e';
import { RouteName } from 'core/Routing/constants/routes';

// utils
import { getRouteByName } from 'core/Routing/utils/getRouteByName';
import { Theme } from 'types/enums';

type TProps = {
  className?: string;
  shouldRedirect?: boolean;
};

const Logo: FC<TProps> = ({ className = '', shouldRedirect = true }) => {
  const { push } = useHistory();
  const { theme } = useContext(Context);

  const onClickHandler = (): void => {
    if (shouldRedirect) {
      push(getRouteByName(RouteName.home));
    }
  };

  return (
    <E2EDataAttribute type={E2EAttribute.logo}>
      {theme === Theme.dark ? (
        <Dark className={className} onClick={onClickHandler} />
      ) : (
        <Light className={className} onClick={onClickHandler} />
      )}
    </E2EDataAttribute>
  );
};

export default Logo;
