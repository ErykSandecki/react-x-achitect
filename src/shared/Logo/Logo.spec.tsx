import React from 'react';
import { createMemoryHistory } from 'history';
import { fireEvent, render } from '@testing-library/react';
import { Router } from 'react-router-dom';

// components
import Logo from './Logo';

// types
import { E2EAttribute } from 'types/e2e';
import { Theme } from 'types/enums';
import { RouteName } from 'core/Routing/constants/routes';

// utils
import { getRouteByName } from 'core/Routing/utils/getRouteByName';
import { getByE2EAttribute } from 'test/testHelpers';

const history = createMemoryHistory({
  initialEntries: [getRouteByName(RouteName.home)],
});

describe('Logo snapshots', () => {
  it('should render in light mode', () => {
    // before
    const { asFragment } = render(
      <Router history={history}>
        <Logo />
      </Router>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render in dark mode', () => {
    // mock
    React.useContext = () => ({ theme: Theme.dark }) as any;

    // before
    const { asFragment } = render(
      <Router history={history}>
        <Logo />
      </Router>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Logo behaviors', () => {
  beforeEach(() => {
    history.push(getRouteByName(RouteName.home));
  });

  it('should redirect on Home Page', () => {
    // before
    const { container } = render(
      <Router history={history}>
        <Logo />
      </Router>,
    );

    // action
    fireEvent.click(getByE2EAttribute(container, E2EAttribute.logo));

    // result
    expect(history.location.pathname).toBe(getRouteByName(RouteName.home));
  });

  it('should not redirect on page when flag is false', () => {
    // before
    const { container } = render(
      <Router history={history}>
        <Logo shouldRedirect={false} />
      </Router>,
    );

    // action
    fireEvent.click(getByE2EAttribute(container, E2EAttribute.logo));

    // result
    expect(history.location.pathname).toBe(getRouteByName(RouteName.home));
  });
});
