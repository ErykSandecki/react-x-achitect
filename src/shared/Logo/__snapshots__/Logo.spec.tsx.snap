// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo snapshots should render in dark mode 1`] = `
<DocumentFragment>
  <div
    class=""
    data-test-logo=""
  />
</DocumentFragment>
`;

exports[`Logo snapshots should render in light mode 1`] = `
<DocumentFragment>
  <div
    class=""
    data-test-logo=""
  />
</DocumentFragment>
`;
